transport/http/roundtrip_test.go

```diff
@@ -62,13 +63,13 @@ func TestRoundTripSuccess(t *testing.T) {
 
 	// start outbound
 	httpTransport := NewTransport()
-	out := httpTransport.NewSingleOutbound(echoServer.URL)
+	var out transport.UnaryOutbound
+	out = httpTransport.NewSingleOutbound(echoServer.URL)
 	require.NoError(t, out.Start(), "failed to start outbound")
 	defer out.Stop()
 
 	// create request
-	hreq, err := http.NewRequest("GET", echoServer.URL, bytes.NewReader([]byte(giveBody)))
-	require.NoError(t, err, "could not create HTTP request")
+	hreq := httptest.NewRequest("GET", echoServer.URL, bytes.NewReader([]byte(giveBody)))
```

Comment:
Making it a server request that has RequestURI set

Reply:
why drop the error check here?

Reply:
changed the method call from `http.NewRequest` that returns a client request to `httptest.NewRequest` that returns a server request. The new method signature does not return an error type.